Packet++/header/DnsResourceData.h

```diff
@@ -5,6 +5,7 @@
 #include "IpAddress.h"
 #include <string>
 #include <stdint.h>
+#include <memory>
```

Comment:
why is this include needed?

Reply:
It's necessary for the following code:

```c++
#if __cplusplus > 199711L || _MSC_VER >= 1800 //Maybe this can be 1600 for VS2010
#define PCPP_SMART_PTR(T) std::unique_ptr<T>
#else
#define PCPP_SMART_PTR(T) std::auto_ptr<T>
#endif
```